File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (int i = 31; i >= 0; --i) {
            int cntBit1 = 0;
            for (int num : nums) {
                if ((num >> i) & 1) {
                    cntBit1++;
                }
            }
            cntBit1 %= 3;
            if (cntBit1) {
                ans |= 1 << i;
            }
        }
        return ans;
    }
};

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for i in range(32):
            cntBit1 = 0
            for num in nums:
                if (num >> i) & 1:
                    cntBit1 += 1
            cntBit1 %= 3
            if cntBit1:
                ans |= 1 << i
               
        return ans - (1 << 32) if ans > (1 << 31) - 1 else ans